### ISSUE TITLE
In the SER_FIQ(on top) model what does nemb signify?

### ISSUE THREAD

**Issue by manisoftwartist** [pterhoer/FaceImageQuality#35]
In your paper, in Section 4 On-top model preparation, you have stated that for networks trained without dropout, we have a new network  "It consist of five layers with nemb/128/512/nemb/nids dimensions". The two intermediate layers have 128 and 512 dimensions. What does nemb mean? Why this network depends on nids

In your code, in class SERFIQ, you have
```
inputs = Input(shape=(25088,))
x = inputs
x = Dropout(0.5)(x, training=True)
x = Dense(512, name="dense", activation="linear")(x)
x = BatchNormalization()(x)
x = Lambda(euclid_normalize)(x)
output = x
```
So the network stricture does not correspond to what you have written in the paper!? 

Why 25088?

**Comment by jankolf** [comment]
Hi @manisoftwartist,
in the paper the specified network is _SER-FIQ (on-top model)_, while the model in our code is _SER-FIQ (same model)_.
The on-top model was created with embeddings taken from either FaceNet or Arcface/Insightface. _n_emb_ is the size of embeddings created by these networks. FaceNet creates 128 dimensional embeddings (_n_emb_ = 128), while Arcface's embeddings are 512 dimensional (_n_emb_ = 512). _n_ids_ is the amount of IDs in your training set.
The same model uses the specified network directly. Therefore, the Dropout layer as well as Dense and Batch Normalization are taken directly from Arcface. As Arcface's Dropout layer has an flattened input dimension of 25088, this number is specified in our code.
I hope this answers your questions.

Best,
Jan

**Comment by manisoftwartist** [comment]
Hello @jankolf

Thanks a lot for your quick response. This partially answered my question and I have more clarity now. 

However, in the SER-FIQ on-top model, I did not understand the choice of binary cross-entropy loss (mentioned in your paper). If the size of the last layer is equal to the identities in the dataset, we should be using categorical cross-entropy? No?

**Comment by pterhoer** [comment]
Hi Mani,

I hope I answered all your questions in the mail exchanges from today :)

Just for all the future issue reader:
SER-FIQ is not a model. It is a concept for measure face image quality that can be applied on any recognition network trained with droput. If your network is not trained with dropout, you can train a recognition network on top of it with at least one fully-connected dropout layer (Just train a recognition network with the pretrained embeddings as input). The network bases (architecture, loss function, data, ...) can be chosen such it fits best for your problem. E.g. categorical cross-entropy might work better then a binary cross-entropy loss in most cases. If you then have a recognition network with dropout, you can simply apply the SER-FIQ methodology as shown in the code example for ArcFace.

Best
Philipp
If you
